meeshkan/serve/mock/response_matcher.py

```diff
@@ -29,15 +23,14 @@ class ResponseMatcher:
 
     def __init__(self, specs: Sequence[OpenAPISpecification]):
         self._specs = specs
+        self._fkr = Faker()
 
-    def match_error(self, msg: str, req: Request) -> Response:
-        return self.default_response(
-            "%s. Here is the full request: host=%s, path=%s, method=%s."
-            % (msg, req.host, req.path, req.method.value)
-        )
+        for spec in specs:
+            storage_manager.add_mock(spec.source, spec.api)
```

Comment:
I think this is a surprising side effect from constructing a `ResponseMatcher` object? Globals are evil so I think it would be very useful if we could somehow make it very explicit where we're modifying them.

Reply:
Agreed! I missed this on my first review, but we should do this either outside of the constructor or make `storage_maanger` a field of the response matcher. Also, this should probably be `storage_store` (see other comment).

Reply:
StorageManager and RestMiddlewareManager aren't globals anymore. Only CallbackManager remains global. I don't know how to avoid this because we have to access it from fixtures.

Reply:
Great job removing dependencies on globals! I think one could make the final global even more explicit by, for example, pushing all added callbacks to an object such as `USER_DEFINED_CALLBACKS` and then injecting that explicitly when creating the `CallbackManager` instance? Just a suggestion though.